cohorts/cohort.py

```diff
@@ -300,22 +295,23 @@ def apply_func(on, col, df):
             (as `self`) along if the function accepts a `cohort` argument.
             """
             if col is None:
-                # Use the function name, or "column" for lambdas, if no
-                # name is provided for the newly created column.
-                col = on.__name__ if not is_lambda(on) else "column"
+                col = func_name(on)
```

Comment:
It's defined outside this function, so should work? That said, don't think anyone calls/needs this, so removing this line.